algorithms/src/snark/marlin/ahp/prover/round_functions/first.rs

```diff
@@ -174,108 +135,38 @@ impl<F: PrimeField, MM: MarlinMode> AHPForR1CS<F, MM> {
         end_timer!(w_poly_time);
-        PoolResult::Witness(LabeledPolynomial::new(label, w_poly, None, Self::zk_bound()))
-    }
-
-    fn calc_z_m(
-        label: impl ToString,
-        evaluations: Vec<F>,
-        constraint_domain: EvaluationDomain<F>,
-        circuit: &Circuit<F, MM>,
-        r: Option<F>,
-    ) -> PoolResult<F> {
-        let should_randomize = MM::ZK && r.is_some();
-        let v_H = constraint_domain.vanishing_polynomial();
-        let label = label.to_string();
-        let poly_time = start_timer!(|| format!("Computing {label}"));
-
-        let evals = EvaluationsOnDomain::from_vec_and_domain(evaluations, constraint_domain);
-
-        let mut poly = evals.interpolate_with_pc_by_ref(&circuit.ifft_precomputation);
-        if should_randomize {
-            poly += &(&v_H * r.unwrap());
-        }
-
-        debug_assert!(
-            poly.evaluate_over_domain_by_ref(constraint_domain)
-                .evaluations
-                .into_iter()
-                .zip(&evals.evaluations)
-                .all(|(z, e)| *e == z),
-            "Label: {label}\n1: {:#?}\n2: {:#?}",
-            poly.evaluate_over_domain_by_ref(constraint_domain).evaluations,
-            &evals.evaluations,
-        );
-
-        let poly_for_opening = LabeledPolynomial::new(label.to_string(), poly, None, Self::zk_bound());
-        if should_randomize {
-            assert!(poly_for_opening.degree() < constraint_domain.size() + Self::zk_bound().unwrap());
-        } else {
-            assert!(poly_for_opening.degree() < constraint_domain.size());
-        }
-
-        let poly_for_committing = if should_randomize {
-            let poly_terms = vec![
-                (F::one(), PolynomialWithBasis::new_lagrange_basis(evals)),
-                (F::one(), PolynomialWithBasis::new_sparse_monomial_basis(&v_H * r.unwrap(), None)),
-            ];
-            LabeledPolynomialWithBasis::new_linear_combination(label, poly_terms, Self::zk_bound())
-        } else {
-            LabeledPolynomialWithBasis::new_lagrange_basis(label, evals, Self::zk_bound())
-        };
-        end_timer!(poly_time);
-
-        PoolResult::MatrixPoly(poly_for_opening, poly_for_committing)
+        Witness(LabeledPolynomial::new(label, w_poly, None, Self::zk_bound()))
     }
 }
 
 #[derive(Debug)]
-pub enum PoolResult<F: PrimeField> {
-    Witness(LabeledPolynomial<F>),
-    MatrixPoly(LabeledPolynomial<F>, LabeledPolynomialWithBasis<'static, F>),
-}
-
-impl<F: PrimeField> PoolResult<F> {
-    fn witness(self) -> Option<LabeledPolynomial<F>> {
-        match self {
-            Self::Witness(poly) => Some(poly),
-            _ => None,
-        }
-    }
-
-    fn z_m(self) -> Option<(LabeledPolynomial<F>, LabeledPolynomialWithBasis<'static, F>)> {
-        match self {
-            Self::MatrixPoly(p1, p2) => Some((p1, p2)),
-            _ => None,
-        }
-    }
-}
+pub struct Witness<F: PrimeField>(LabeledPolynomial<F>);
```

Comment:
We can just make this a `pub type Witness<F> = ...`

Reply:
080107284